core/services/pipeline/common.go

```diff
@@ -237,25 +237,11 @@ func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 
 // MarshalJSON implements custom marshaling logic
 func (js JSONSerializable) MarshalJSON() ([]byte, error) {
-	if !js.Valid {
+	if !js.Valid || js.Val == nil {
```

Comment:
Will this lead to literal `null` string being written to the database?

Reply:
Correct, this is valid JSON and this is handled by `UnmarshalJSON`, producing `{Val: nil, Valid: false}`.